tests/tracking/request_header/test_databricks_request_header_provider.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+from unittest import mock
+
+from mlflow.tracking.request_header.databricks_request_header_provider import (
+    DatabricksRequestHeaderProvider,
+)
+
+
+def test_databricks_cluster_run_context_in_context():
+    with mock.patch("mlflow.utils.databricks_utils.is_in_cluster") as in_cluster_mock:
+        assert DatabricksRequestHeaderProvider().in_context() == in_cluster_mock.return_value
+
+
```

Comment:
Can we add a couple of tests for the case where we're in multiple contexts, ensuring that all relevant information is captured: one for the case where we have cluster ID + notebook info available and one for the case where we have cluster ID + job info available?

Reply:
Do you mean to update the `in_context()` to be an OR of the 3 environments and then update this test accordingly?

Reply:
Don't the existing tests add whatever info we have available?

Reply:
They don’t verify that we add cluster id and notebook info if both are available. I.e. I’m pretty sure that all tests would pass if we used the kind of if / elif logic that mohamad asked about above.